sdk/ai/azure-ai-projects/azure/ai/projects/_patch.py

```diff
@@ -0,0 +1,290 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Customize generated code here.
+
+Follow our quickstart for examples: https://aka.ms/azsdk/python/dpcodegen/python/customize
+"""
+import uuid
+from os import PathLike
+from pathlib import Path
+from typing import List, Any, Union, Dict
+from typing_extensions import Self
+from azure.core.credentials import TokenCredential
+from azure.core import PipelineClient
+from azure.core.pipeline import policies
+from ._configuration import AIProjectClientConfiguration
+from ._serialization import Deserializer, Serializer
+from .operations import AgentsOperations, ConnectionsOperations, EvaluationsOperations, TelemetryOperations
+from ._client import AIProjectClient as ClientGenerated
+from .operations._patch import InferenceOperations
+
+
+class AIProjectClient(ClientGenerated):
+
+    def __init__(
+        self,
+        endpoint: str,
+        subscription_id: str,
+        resource_group_name: str,
+        project_name: str,
+        credential: "TokenCredential",
+        **kwargs: Any,
+    ) -> None:
+        # TODO: Validate input formats with regex match (e.g. subscription ID)
+        if not endpoint:
+            raise ValueError("endpoint is required")
+        if not subscription_id:
+            raise ValueError("subscription_id ID is required")
+        if not resource_group_name:
+            raise ValueError("resource_group_name is required")
+        if not project_name:
+            raise ValueError("project_name is required")
+        if not credential:
+            raise ValueError("Credential is required")
+        if "api_version" in kwargs:
+            raise ValueError("No support for overriding the API version")
+        if "credential_scopes" in kwargs:
+            raise ValueError("No support for overriding the credential scopes")
```

Comment:
@kristapratico At the moment the client library talks to several services, so it does not make sense to provide api_version and credential_scopes in client constructor, as we don't know which service to apply those. With the move to new unified dataplane APIs on a single service, this code will be written and it will look more like a typical Azure SDK client. So this check is temporary.